Makefile

```diff
@@ -24,4 +25,127 @@ lint: $(GOLANGCI_LINT)
 fmt: $(GOIMPORTS)
 	$(GOIMPORTS) -w -local github.com/Azure/ARO-HCP $(shell go list -f '{{.Dir}}' -m | xargs)
 
-.PHONY: all clean lint test fmt
+#
+# Infra
```

Comment:
i'm intrigued by the idea of generating targets of the top level makefile
but i would argue that it should be mostly about the deploy target of each service and not the many others that are mostly used by the individual service team.

> This might also remove the need for scripts like svc-deploy.sh
the placement of a component on an individual cluster is not really a service team / Makefile owner decision. but we can definitely think of changing this. but it is an implementation detail and can be postponed to another PR

Reply:
That's fair. If we just want `deploy` proxied, we could consider `%/deploy` as a target alternative here so only that command is proxied up using a directory structure - the target content would be the same as above.

Reply:
As always being careful with arcane `make` invocations is likely a good idea. Implicit rules and `%` are just useful enough to be worth it!

Reply:
@geoberle not sure if this is useful or not, but I have had great success with actually outputting the results of `az deployment` calls `-o json` into files in some `.gitignore`-d directory like `_output`. No need to have `.PHONY` targets and complex `make` cascades can leverage `make` itself to be more efficient on re-run.